Update tweakreg docs to describe fit to GAIA (#5811)
spacetelescope/jwst: docs/jwst/tweakreg/README.rst

---------
The source catalogs for each input image are compared to each other
and linear (affine) coordinate transformations that align these
catalogs are derived.  This fit insures that all the input images
are aligned relative to each other.  This step produces a combined
source catalog for the entire set of input images as if they were
combined into a single mosaic.

> An astrometric reference catalog then gets generated by querying

Reviewer: I believe there's a step param that controls whether or not the alignment to GAIA is done, hence this description should be prefixed with something like "If step parameter <blee> is True, an astrometric reference catalog ..."

Author: Resolved in efc699a.

---------
The source catalogs for each input image are compared to each other
and linear (affine) coordinate transformations that align these
catalogs are derived.  This fit insures that all the input images

> are aligned in a 'relative' manner.  This results in a combined

Reviewer 2: ```suggestion
are aligned relative to each other.  This step produces a combined
```

Author: Addressed in e6ae123.

source catalog for the entire set of input images as if they were
combined into a single mosaic.

An astrometric reference catalog then gets generated by querying
a GAIA-based astrometric catalog web service for all astrometrically
measured sources in the combined field-of-view of the set of input
images. This catalog gets generated from the catalogs available

> through the GSSS STScI web service at:

Reviewer 2: For this to work, you need to cross reference it. For example:
```suggestion
through the GSSS STScI web service at `GSSSCATALOGS`_.
```

Author: Addressed in e6ae123.

> .. _GSSSCATALOGS: http://gsss.stsci.edu/webservices/vo/CatalogSearch.aspx

Reviewer 3: This does not render correctly. Perhaps the formatting should be removed.

> The combined source catalog derived from the input images

Reviewer 2: ```suggestion
The combined source catalog derived in the first step
```

Author: Addressed in e6ae123.

> then gets cross-matched and fit to this astrometric reference catalog
> to align the input images to the astrometric reference catalog's
> coordinate system.  The pipeline initially supports fitting to the

Reviewer 2: I find this slightly not very clear. I would suggest eliminating "to align ..." - this is well explained in the last sentences.
```suggestion
then gets cross-matched and fit to this astrometric reference catalog.
The pipeline initially supports fitting to the
```

Author: Addressed in e6ae123.

GAIADR2 catalog, with the option to select the GAIADR1 instead.

> The results of this one fit then get applied to all the

Reviewer 2: ```suggestion
The results of this one fit then get back-propagated to all the
```

Author: Addressed in e6ae123.

* ``sigma``: A positive `float` indicating the clipping limit, in sigma units,
  used when performing fit. (Default=3.0)

**Astrometric fitting parameters:**

* ``align_to_gaia``: A boolean indicating whether or not to fit to
  perform the fit to an astrometric catalog. (Default=False)

* ``gaia_catalog``: String indicating what astrometric catalog should be used.

>   Initially supported options:{'GAIADR2', 'GAIADR1'}. (Default='GAIADR2')

Reviewer 2: Since this is a "readme" (as opposite to API documentation), a descriptive (natural language) could be easier to the reader. I suggest dropping curly brackets.
```suggestion
  Initially supported options: 'GAIADR2' and 'GAIADR1'. (Default='GAIADR2')
```

Author: Addressed in e6ae123.
